Fatras/include/ActsFatras/Physics/PhotonConversion/PhotonConversion.hpp

```diff
@@ -0,0 +1,350 @@
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in] gammaMom4 The momentum four vector of the photon
+  ///
+  /// @return The direction vector of the child particle
+  template <typename generator_t>
+  Particle::Vector3 childDirection(generator_t& generator,
+                                   const Particle::Vector4& gammaMom4) const;
+
+  /// Helper methods for momentum evaluation
+  /// @note These methods are taken from the Geant4 class
+  /// G4PairProductionRelModel
+  Scalar screenFunction1(Scalar delta) const;
+  Scalar screenFunction2(Scalar delta) const;
+
+  static constexpr Scalar m_Z = 13.;  // Aluminium
+
+  /// Fractions
+  static constexpr Scalar m_oneOverThree = 1. / 3.;
+  static constexpr Scalar m_nineOverSeven = 9. / 7.;
+};
+
+inline Particle::Scalar PhotonConversion::screenFunction1(Scalar delta) const {
+  // Compute the value of the screening function 3*PHI1(delta) - PHI2(delta)
+  return (delta > 1.4) ? 42.038 - 8.29 * log(delta + 0.958)
+                       : 42.184 - delta * (7.444 - 1.623 * delta);
+}
+
+inline Particle::Scalar PhotonConversion::screenFunction2(Scalar delta) const {
+  // Compute the value of the screening function 1.5*PHI1(delta)
+  // +0.5*PHI2(delta)
+  return (delta > 1.4) ? 42.038 - 8.29 * log(delta + 0.958)
+                       : 41.326 - delta * (5.848 - 0.902 * delta);
+}
+
+template <typename generator_t>
+std::pair<Particle::Scalar, Particle::Scalar>
+PhotonConversion::generatePathLimits(generator_t& generator,
+                                     const Particle& particle) const {
+  /// This method is based upon the Athena class PhotonConversionTool
+
+  // Fast exit if not a photon or the energy is too low
+  if (particle.pdg() != 22 ||
+      particle.absoluteMomentum() < 2. * childMomentumMin)
+    return std::make_pair(std::numeric_limits<Scalar>::infinity(),
+                          std::numeric_limits<Scalar>::infinity());
+
+  // Use for the moment only Al data - Yung Tsai - Rev.Mod.Particle Physics Vol.
+  // 46, No.4, October 1974 optainef from a fit given in the momentum range 100
+  // 10 6 2 1 0.6 0.4 0.2 0.1 GeV
+
+  //// Quadratic background function
+  //  Double_t fitFunction(Double_t *x, Double_t *par) {
+  //  return par[0] + par[1]*pow(x[0],par[2]);
+  // }
+  // EXT PARAMETER                                   STEP         FIRST
+  // NO.   NAME      VALUE            ERROR          SIZE      DERIVATIVE
+  //  1  p0          -7.01612e-03   8.43478e-01   1.62766e-04   1.11914e-05
+  //  2  p1           7.69040e-02   1.00059e+00   8.90718e-05  -8.41167e-07
+  //  3  p2          -6.07682e-01   5.13256e+00   6.07228e-04  -9.44448e-07
+  constexpr Scalar p0 = -7.01612e-03;
+  constexpr Scalar p1 = 7.69040e-02;
+  constexpr Scalar p2 = -6.07682e-01;
+
+  // Calculate xi
+  const Scalar xi = p0 + p1 * pow(particle.absoluteMomentum(), p2);
```

Comment:
Note: `thirdparty/dfelibs` has a library for numerically stable and efficient evaluation of polynomials.

Reply:
Good to know but due to the value of `p2` I cannot see how this is useful for this problem. Or did I miss the right file/method?

Reply:
Its the `dfe_poly.hpp` header. I think it can be useful regardless of the leading value since you get rid of the `pow(...)` call and reduce the number of opertaions.

 I just wanted to mention it; no need to change that here.

Reply:
But that's only applicable if `p2` is an integer , no?

Reply:
Ahh, I misread the function. I though it was just a regular polynomial and not `a + b * x^c`. In that case, all my comments do not apply of course. My apologies.